server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -566,6 +566,9 @@ public void updateShardState(
                     : "a primary relocation is completed by the cluster-managerr, but primary mode is not active " + currentRouting;
 
                 changeState(IndexShardState.STARTED, "global state is [" + newRouting.state() + "]");
+                if (indexSettings.isSegRepEnabled()) {
```

Comment:
Can we add a comment here on why this is required? I think this is so new primary shards will push to replicas after relocation rather than waiting for more docs to get indexed?

Reply:
Yes, new primary shards will push to replicas after relocation rather than waiting for more docs to get indexed